pkg/runtime/bulksub_resiliency.go

```diff
@@ -0,0 +1,63 @@
+package runtime
+
+import (
+	"context"
+	"sync/atomic"
+
+	"github.com/cenkalti/backoff/v4"
+	"github.com/pkg/errors"
+
+	"github.com/dapr/components-contrib/pubsub"
+	"github.com/dapr/dapr/pkg/resiliency"
+	"github.com/dapr/dapr/utils"
+)
+
+// ApplyBulkSubscribeResiliency applies resiliency support to bulk subscribe. It tries to filter
+// out the messages that have been successfully processed and only retries the ones that have failed
+func (a *DaprRuntime) ApplyBulkSubscribeResiliency(bulkSubCallData *bulkSubscribeCallData,
+	psm pubsubBulkSubscribedMessage, deadLetterTopic string, path string, policyDef *resiliency.PolicyDefinition,
+	rawPayload bool, envelope map[string]interface{},
+) (*[]pubsub.BulkSubscribeResponseEntry, error) {
+	bscData := *bulkSubCallData
+	ctx := bscData.ctx
+	var subscribedEntries atomic.Pointer[[]pubSubMessage]
+	subscribedEntries.Store(&psm.pubSubMessages)
+	policyRunner := resiliency.NewRunnerWithOptions(
+		ctx, policyDef, resiliency.RunnerOpts[*[]pubsub.BulkSubscribeResponseEntry]{
+			Accumulator: func(bsre *[]pubsub.BulkSubscribeResponseEntry) {
+				for _, v := range *bsre {
+					// add to main bulkResponses
+					index := (*bscData.entryIdIndexMap)[v.EntryId]
```

Comment:
Is this guaranteed to be found, or is there a chance that index will be the zero value and so `(*bscData.bulkResponses)[index]` would be nil? should there be a nil-check?

Reply:
Yeah, this is guaranteed to be found. But, still it doesn't harm to put an extra check here, so putting an explicit check here as well.